publication-commons/src/main/java/no/unit/nva/publication/service/impl/UpdateResourceService.java

```diff
@@ -144,18 +149,61 @@ public ImportCandidate updateImportCandidate(ImportCandidate importCandidate) th
         throw new BadRequestException("Can not update already imported candidate");
     }
 
-    public void unpublishPublication(Publication publication) {
+    public void unpublishPublication(Publication publication,
+                                     Stream<TicketEntry> existingTicketStream,
+                                     UnpublishRequest unpublishRequest) {
         publication.setStatus(UNPUBLISHED);
         publication.setDuplicateOf(publication.getDuplicateOf());
         publication.setModifiedDate(clockForTimestamps.instant());
         var resource = Resource.fromPublication(publication);
 
+
+        var existingTicketTransactionsItems =
+            existingTicketStream
+                .filter(this::isPendingTicket)
+                .map(this::setToNotApplicable)
+                .map(Entity::toDao)
+                .map(dao -> (TicketDao) dao)
+                .map(this::createPutTransactionItems)
+                .toList();
+        var unpublishinRequestTransationItems =
+            new UnpublishRequestDao(unpublishRequest).createInsertionTransactionRequest().getTransactItems();
         var updateResourceTransactionItem = updateResource(resource);
 
-        var request = new TransactWriteItemsRequest().withTransactItems(updateResourceTransactionItem);
+        var transactionItems = new ArrayList<>(existingTicketTransactionsItems);
+        transactionItems.add(updateResourceTransactionItem);
+        transactionItems.addAll(unpublishinRequestTransationItems);
+
+        var request = new TransactWriteItemsRequest().withTransactItems(transactionItems);
         sendTransactionWriteRequest(request);
     }
```

Comment:
This is esoteric.
(I have read and understand the code, except this line -> `publication.setDuplicateOf(publication.getDuplicateOf());` )

Not complaining about your code, It is the library we are using that seems to suck.

Reply:
```
        var request = new TransactWriteItemsRequest()
            .withTransactItems(existingTicketTransactionsItems)
            .withTransactItems(updateResourceTransactionItem)
            .withTransactItems(unpublishinRequestTransationItems);        
            
        sendTransactionWriteRequest(request);
```
isn't this easier to read?

Reply:
It is easier to read, however "withTransactionItems" is a setter, it doesn't append to existing transactionsItems.

Reply:
The `publication.setDuplicateOf(publication.getDuplicateOf()); ` is from the old version of the code. I have noe idea why it is there. Removing; well spotted.

Reply:
"This is esoteric." -> I agree, but we are deep into the implementation of resource service, a tight coupling to dynamoDB and a certain level of domain knowledge is expected of anyone venturing into this part of the code. The tests are the documentation of the functionality. If they don't reveal the code's intent, the tests are poorly written.

Reply:
Attempted to make the code more "clean".